### PR TITLE
TosaToTensor: Support reshape on tensors of unsigned integer

### PR DESCRIPTION
**Pull request by mgehre-amd**
This adds 
- `mlir::tosa::populateTosaToLinalgTypeConversion` which converts tensors of unsigned integers into tensors of signless integers
- modifies the `tosa.reshape` lowering in TosaToTensor to use the type converter correctly

I choose to implement the type converter in `mlir/Conversion/TosaToLinalg/TosaToLinalg.h` instead of `mlir/Conversion/TosaToTensor/TosaToTensor.h` because I need the same type converter in the TosaToLinalg lowerings (future PR).
Alternatively, I could duplicate the type converter so it exists both in TosaToLinalg and TosaToTensor. Let me know if you prefer that.

### CODE REVIEW

**Comment by mgehre-amd** [comment]
This is how TosaToLinalg would use the type converter: #91734

**Review by leandron — APPROVED** [review]
LGTM, thanks @mgehre-amd

**Comment by sjarus** [comment]
Would it make sense to put the type converter inside the TOSA dialect itself as a utility function ? This would make it broadly useful to pathways that may not use Linalg.

File: mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp [hunk 1]
```diff
@@ -2617,3 +2617,37 @@ void mlir::tosa::populateTosaToLinalgConversionPatterns(
       TileConverter>(patterns->getContext());
   // clang-format on
 }
+
+void mlir::tosa::populateTosaToLinalgTypeConversion(TypeConverter &converter) {
```
Comment (sjarus) [inline mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp:2621]: Looks good in general but I'm wondering if a common case type converter could sit as a utility function within the source dialect itself ?

Comment (mgehre-amd) [reply mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp]: Good idea, I moved this to `mlir/lib/Dialect/Tosa/Transforms/TosaTypeConverters.cpp`

**Review by sjarus — APPROVED** [review]
Thanks for refactoring this!

**Pull request merged**